src/algorithm/centroid.rs

```diff
@@ -18,62 +19,75 @@ pub trait Centroid {
     /// println!("Centroid {:?}", linestring.centroid());
     /// ```
     ///
-    fn centroid(&self) -> Option<Point>;
+    fn centroid(&self) -> Option<Point<T>>;
 }
 
-impl Centroid for LineString {
+impl<T> Centroid<f64> for LineString<T>
+    where T: Num + Copy + ToPrimitive
+{
     ///
     /// Centroid on a LineString is the mean of the middle of the segment
     /// weighted by the length of the segments.
     ///
-    fn centroid(&self) -> Option<Point> {
+    fn centroid(&self) -> Option<Point<f64>> {
         let vect = &self.0;
         if vect.is_empty() {
             return None;
         }
         if vect.len() == 1 {
-            Some(Point::new(vect[0].lng(), vect[0].lat()))
+            Some(Point::new(vect[0].x().to_f64().unwrap(),
+                            vect[0].y().to_f64().unwrap()))
         } else {
-            let mut sum_x = 0.;
-            let mut sum_y = 0.;
-            let mut total_length = 0.;
+            let mut sum_x : f64 = 0.;
+            let mut sum_y : f64 = 0.;
+            let mut total_length : f64 = 0.;
```

Comment:
This could be made generic by doing:

```rust
let mut total_length = T::zero();
```

http://rust-num.github.io/num/num/trait.Zero.html